Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1175,6 +1157,93 @@ private extension PurchasesOrchestrator {
         }
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func sk2PromotionalOffer(forProductDiscount productDiscount: StoreProductDiscountType,
+                             discountIdentifier: String,
+                             product: StoreProductType,
+                             subscriptionGroupIdentifier: String,
+                             completion: @escaping @Sendable (Result<PromotionalOffer, PurchasesError>) -> Void) {
+        self.transactionFetcher.fetchLastVerifiedTransaction { transaction in
+            guard let transaction = transaction, let jwsRepresentation = transaction.jsonRepresentation  else {
+                // Promotional offers require existing purchases.
+                // Fail early if there are no transactions
+                completion(.failure(ErrorUtils.ineligibleError()))
```

Comment:
👍🏻 
Let's add tests for this too.

The good thing is that our current integration tests will cover this as well, but I'd cover it in `PurchasesOrchestratorTests` too.